### PR TITLE
Add keymap for keys with shift modifier

### PR DESCRIPTION
**Pull request by nxnfufunezn**

### CODE REVIEW

File: src/keyboard/src/lib.rs [hunk 1]
```diff
@@ -1,16 +1,56 @@
...
+}
 
 pub struct Keyboard;
 
 impl Keyboard {
 	pub fn handle_keys(&self, scancode: usize) {
 		if scancode <= 59 {
-			kprint!("{}", KBDUS[scancode] as char);
+			let state = STATE.lock();
+			if state.shift ^ state.caps {
```
Comment (WilsonGiese) [inline src/keyboard/src/lib.rs]: Is this is the expected behavior for caps & shift being pressed? I don't really know if there is a set standard, but on Mac OS X when caps lock is on the key pressed is always caps.

Comment (nxnfufunezn) [inline src/keyboard/src/lib.rs]: IIRC mac caps lock work differently that's why...

Comment (WilsonGiese) [inline src/keyboard/src/lib.rs]: Heh, weird.

File: src/keyboard/Cargo.toml [hunk 2]
```diff
@@ -5,3 +5,4 @@ authors = ["nxnfufunezn <nxnfufunezn@gmail.com>"]
 
 [dependencies]
 vga = { path = "../vga" }
+spin = "0.3.5"
```
Comment (steveklabnik) [inline src/keyboard/Cargo.toml]: We're using 0.4.2 now, mind updating?

**Comment by steveklabnik** [comment]
This needs to be rebased due to lockfiles, would you mind doing so? Thanks!

**Comment by nxnfufunezn** [comment]
@steveklabnik done!

**Comment by steveklabnik** [comment]
Thanks! :D

**Pull request merged**